### PR TITLE
fix: property pane table binding issue fixed

### LINKED ISSUE #25432: [Bug]:  API when binded to a table does not dispaly the API in Table Data of property pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When user binds an API to a table it is observed that the Table data of property pane remains empty 

### Steps To Reproduce

1) Create an API 
2) Run the. API
3) Create a Binding to a table 
observe user is navigated to Canvas
4) Observe the property pane of Table
5) Table data field remains empty

Screenshot 2023-07-18 at 12 05 42 PM

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by sneha122**
## Description

When REST API with which returns response in non array format is bound to table widget, We are not able to see table's data property populated with that api name. This PR fixes the issue
Screenshot 2023-08-08 at 2 51 12 PM

#### PR fixes following issue(s)
Fixes #25432 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- Bug fix (non-breaking change which fixes an issue)
>
>
>
- [x] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/SuggestedWidgets_spec.js [hunk 1]
```diff
@@ -3,16 +3,21 @@ import {
   dataSources,
   table,
   agHelper,
+  apiPage,
+  assertHelper,
+  entityExplorer,
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/SuggestedWidgets_spec.js:8]: Import `entityItems` from ObjectsCore itself.

**Review by sbalaji1192 — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/SuggestedWidgets_spec.js [hunk 2]
```diff
@@ -31,4 +36,26 @@ describe("Check Suggested Widgets Feature in auto-layout", function () {
       expect(cellData).to.eq("5");
     });
   });
+
+  it("Bug 25432. Table widget bound to API with mock api url", () => {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/SuggestedWidgets_spec.js:40]: We already do this in ClientSide/BugTests/ApiBugs_Spec.ts [1st test] - Can we consider adding validation to that case?

File: app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/SuggestedWidgets_spec.js [hunk 3]
```diff
@@ -3,16 +3,21 @@ import {
...
   table,
   agHelper,
+  apiPage,
+  assertHelper,
+  entityExplorer,
 } from "../../../../support/Objects/ObjectsCore";
 import { Widgets } from "../../../../support/Pages/DataSources";
 import { featureFlagIntercept } from "../../../../support/Objects/FeatureFlags";
+import { PROPERTY_SELECTOR } from "../../../../locators/WidgetLocators";
+import { EntityItems } from "../../../../support/Pages/AssertHelper";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/MobileResponsiveTests/SuggestedWidgets_spec.js:13]: Are these imports still needed here?

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/ApiBugs_Spec.ts [hunk 4]
```diff
@@ -25,16 +28,34 @@ describe("API Bugs", function () {
...
     apiPage.CreateAndFillApi(
       tedTestConfig.dsValues[tedTestConfig.defaultEnviorment].mockApiUrl,
       "Api1",
     );
     apiPage.RunAPI();
-
     dataSources.AddSuggestedWidget(Widgets.Table);
-
     debuggerHelper.AssertErrorCount(0);
+    agHelper.AssertElementExist(
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/ApiBugs_Spec.ts:40]: `AssertPropertiesDropDownCurrentValue` dint help here?

**Review by Aishwarya-U-R — COMMENTED** [review]
Approving Cypress changes.

**Review by Aishwarya-U-R — APPROVED** [review]

**Pull request merged**